macros/database_specific_helpers/get_relation.sql

```diff
@@ -0,0 +1,14 @@
+{% macro get_relation(get_relation_name) %}
+    {% if execute %}
+        {% set model_get_relation_node = graph.nodes.values() | selectattr('name', 'equalto', get_relation_name) | first %}
+        {% set relation = api.Relation.create(
+            database = model_get_relation_node.database,
+            schema = model_get_relation_node.schema,
+            identifier = model_get_relation_node.alias
+        )
+        %}
+        {% do return(relation) %}
+    {% else %}
+        {% do return(api.Relation.create()) %}
```

Comment:
Does this need to be passed anything in order to create the correct Relation? e.g. the relation_name?

Reply:
I don't think it does as [everything is optional](https://docs.getdbt.com/reference/dbt-classes#relation), although not sure what it will create if they aren't passed it? Fwiw, this was copied and pasted directly from the `upload_results.sql` file as there were multiple macros in there.